### PR TITLE
Room finder command + probably better course fetching

### PR DESCRIPTION
**Pull request by andesyv**
- I rewrote a bunch of the code that fetches course information to make it more efficient on runtime, and probably fetch more information.
 - I also added a command to search for a room. You can for instance do `!where 209N1` or `!where "auditorium 2"` and it will link you directly to a map, or to a UiB page containing room info + a map.

### CODE REVIEW

File: src/commands/uib/exams.ts [hunk 1]
```diff
@@ -27,44 +26,51 @@ class ExamCommand extends Command {
...
     }
 
-    const course = await getExams(inputSubject, message);
+    const course = await getCourse(inputSubject, message);
     if (course === undefined) {
       return await message.reply(`Sorry, no course with the code ${inputSubject} found... try again`);
     }
 
+    if (course.exams?.length === 0 ?? true) {
+      return await message.reply(`It doesn't seem to be any exams for ${inputSubject}.`);
```
Comment (sondr3) [inline src/commands/uib/exams.ts:41]:
```suggestion
      return await message.reply(`There doesn't seem to be any exams for ${inputSubject}.`);
```

File: src/commands/uib/exams.ts [hunk 2]
```diff
@@ -27,44 +26,51 @@ class ExamCommand extends Command {
...
-      }
-      if (Object.prototype.hasOwnProperty.call(exam, "duration")) {
-        fields.push({ name: "Varighet", value: exam.duration, inline: true });
-      }
-      // if (Object.prototype.hasOwnProperty.call(exam, "location")) {
-      //   fields.push({ name: "Location", value: exam.location, inline: true });
-      // }
-      if (Object.prototype.hasOwnProperty.call(exam, "system")) {
-        fields.push({ name: "Eksamensystem", value: exam.system, inline: true });
+    if (course.exams) {
```
Comment (sondr3) [inline src/commands/uib/exams.ts:56]: Har denne plutselig begynt å kræsje her eller?

File: src/commands/uib/room.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  run = async (message: CommandoMessage, { room }: { room: string }): Promise<Message | Message[]> => {
+    if (room === "") {
+      return await message.reply("You need to specify the room to locate");
+    }
+
+    const data = await getPersistentData();
+    if (data === undefined) {
+      return await message.reply(`Sorry, cannot find any rooms atm.`);
```
Comment (sondr3) [inline src/commands/uib/room.ts:34]:
```suggestion
      return await message.reply(`Sorry, cannot find any rooms.`);
```

**Review by sondr3 — COMMENTED** [review]
Litt småting, ellers LGTM :shipit:

File: src/commands/uib/exams.ts (hunk 2 above)
Comment (andesyv) [reply src/commands/uib/exams.ts]: Jeg endret variablen til en nullable, ettersom den ikke alltid har en verdi og det virket mer ryddig enn en tom liste. Gjorde koden litt mer tydelig for å reflektere det.

**Review by sondr3 — APPROVED** [review]

**Pull request merged**